### PR TITLE
Movie App (JavaScript capstone project)

### PR DESCRIPTION
**Pull request by Adholah96**
# **In this project, We completed the following:** 
# Group Task : 

- [ ] Find External API where we picked [MovieApi](https://api.tvmaze.com/shows) and [InvolvementApi](https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/)
- [ ] Set up the project with webpack and jest.
- [ ] Designed and included both header and footer. 
- [ ] Chose favorable font style, color scheme, and images

# Student A: @Adholah96  : 

- [ ]  Display a list of movies consumed from the API on the App. 
- [ ] Display the number of likes of each movie on the Homepage. 
- [ ] Create a like feature that increments likes when triggered. 
- [ ] Add a counter that tracks all the movies added to the DOM. 
- [ ] Design a jest test that tests the movie counter. 

# Student B: @Kevin-Mena : 

- [ ]  Display comments for every movie displayed on the DOM. 
- [ ] Display comment pop for each and every movie. 
- [ ] create a function that adds comments to every movie. 
- [ ] Add comment counter and display on the comment popup. 
- [ ] Design a jest test that tests the comment counter.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  const animes = await fetchAnime();
+  displayAnime(animes);
+  likedAnimes();
+  const likes = await getLikes();
+  showLike(likes);
+
+  const movies = document.querySelectorAll('.anime-holder');
+  const allMovies = movies.length;
+  const totalHolder = document.querySelector('.totality');
+  totalHolder.innerHTML = allMovies;
```
Comment (leonardodiasb) [inline src/index.js:34]: - [ ] This is the counter functionality that counts the number of movies/series being displayed on the Home page. This function works fine, but it should be implemented in a separate module so you can import it into your test file. Kindly create a new module for it so you can import it here and on your test file.

File: src/modules/itemsCounter.test.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        `;
+
+    const movies = document.querySelectorAll('.anime-holder');
+    const allMovies = movies.length;
+    const totalHolder = document.querySelector('.totality');
+    totalHolder.innerHTML = allMovies;
+
+    expect(totalHolder.innerHTML).toEqual('3');
+  });
+});
```
Comment (leonardodiasb) [inline src/modules/itemsCounter.test.js:19]: - [ ] Your 'itemsCounter' test file is currently not importing the function you created for counting the number of movies/animes on your Home page. Kindly import the function and use it in your test file. Currently, you are only calculating the number of movies with the `allMovies` const, which is not testing any function being used in your application. Please, remember to test edge cases, like if there is no movie/series on your page for example.

File: src/modules/displayComment.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  const shwComments = document.querySelector('.comments');
+
+  commentData.forEach((com) => {
+    shwComments.innerHTML += `
+        <p class="addedComment"><b>UserName</b>: ${com.username}</br></br> <b>Comment</b>: </br> ${com.comment}</p></br><hr/>
+    `;
+  });
+  comments.appendChild(shwComments);
+  totalCount.textContent = counter;
+};
```
Comment (leonardodiasb) [inline src/modules/displayComment.js:18]: - [ ] Like the items counter, the comments counter should be a function implemented in a separate module where you can import from different files like the test file. Kindly extract the function that counts the comments and implement it in a new module.

File: src/modules/commentCounter.test.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    document.querySelector = jest.fn(() => mockCommentsContainer);
+  });
+
+  it('should increment the comment counter', () => {
+    const initialCounter = mockCommentsNumber.textContent;
+    includeUI('testUser', 'testComment');
+    const updatedCounter = mockCommentsNumber.textContent;
+    expect(parseInt(updatedCounter, 10)).toBe(parseInt(initialCounter, 10) + 1);
+  });
+});
```
Comment (leonardodiasb) [inline src/modules/commentCounter.test.js:21]: - [ ] Kindly refactor your test file for testing the comment counter. Currently, it is not importing the function that calculates the number of comments of a given movie/series, and the mock you created is returning undefined objects when the tests run. You could create a mock HTML as you created in the other test file and import the function that counts the number of comments and test if the number is correct. Please, remember to test edge cases if there are no comments to count.
 - 
image

**Review by leonardodiasb — CHANGES_REQUESTED** [review]
# Required Changes ♻️ ⚠️ 🚦 

Hi @Adholah96, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

mario_ladders

## Good points 👍 

 - No linter errors. ✔️
 - API calls working fine. ✔️ 
 - Good use of ES6 best practices. ✔️ 

## Aspects to improve ♻️

- [x] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zieeco — APPROVED** [review]
# status: APPROVED :green_circle: :100: 

Hi, @Team, 


68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f78543078657a514755357843444a754350652f67697068792e676966


Good job so far 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⚡ 

✅ Descriptive PR 👍 
✅ Nice design 🔥 
✅ Good commit messages 👍 
✅ Linters are passing 👏 


## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
